refiner/app/services/refine.py

```diff
@@ -318,52 +326,121 @@ def refine_eicr(
         )
 
 
+def build_condition_eicr_pairs(
+    original_xml_files: XMLFiles,
+    reportable_conditions: list[dict[str, Any]],
+) -> list[dict[str, Any]]:
+    """
+    Generate pairs of reportable conditions and fresh XMLFiles copies.
+
+    Each reportable condition gets its own copy of the XMLFiles object so that
+    condition-specific refinement can be performed independently.
+
+    Args:
+        original_xml_files: The original XMLFiles object containing eICR and RR.
+        reportable_conditions: List of reportable condition objects with 'code' keys.
+
+    Returns:
+        list[dict[str, Any]]: A list of dictionaries, each containing a
+                             `reportable_condition` and `xml_files`.
+    """
+
+    condition_eicr_pairs = []
+
+    for condition in reportable_conditions:
+        # create fresh XMLFiles for each condition to ensure complete isolation
+        condition_xml_files = XMLFiles(original_xml_files.eicr, original_xml_files.rr)
+        condition_eicr_pairs.append(
+            {
+                "reportable_condition": condition,
+                "xml_files": condition_xml_files,
+            }
+        )
+
+    return condition_eicr_pairs
+
+
+# NOTE:
+# SECTION PROCESSING (Core refinement logic)
+# =============================================================================
+
+
 def _process_section(
     section: _Element,
     combined_xpath: str,
     namespaces: dict = {"hl7": "urn:hl7-org:v3"},
+    section_config: dict | None = None,
+    version: str = "1.1",
```

Comment:
Should we make this a Literal too?
```suggestion
    version: Literal["1.1"] = "1.1",
```

Reply:
yeah, i think changing to a Literal is actually the right thing to do